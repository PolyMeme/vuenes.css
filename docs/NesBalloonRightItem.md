# nes-balloon-right-item 

Vue Component implementation of the Nes Balloon in the Right Side 

- **author** - Jesus Fernando Alvarez Franco 
- **license** - MIT 

## props 

- `message` ***String*** (*optional*) `default: Empty String` 

  Define the message of the balloon message 

