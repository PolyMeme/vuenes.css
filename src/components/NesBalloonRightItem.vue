<template>
  <div class="message -right">
    <div class="nes-balloon from-right">
      <p>{{ message }}</p>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    message: {
      type: String,
      default: () => ''
    }
  }
}
</script>

<style lang="scss">
@import "~typeface-press-start-2p/index.css";
@import "~nes.css/scss/base/index";
@import "~nes.css/scss/elements/balloons";
@import "~nes.css/scss/utilities/index";
</style>
