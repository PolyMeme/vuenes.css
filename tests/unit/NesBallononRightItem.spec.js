import { mount } from '@vue/test-utils'
import NesBalloonRightItem from '../../src/components/NesBalloonRightItem.vue'

describe('NesBalloonRightItem.vue', () => {
  it('should render empty message', () => {
    const wrapper = mount(NesBalloonRightItem)
    expect(wrapper.find('div.message.-right>div.nes-balloon.from-right>p').text()).toBe('')
  })

  it('should match snapshot', () => {
    const wrapper = mount(NesBalloonRightItem, {
      propsData: {
        message: 'This is a message from the right!'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
