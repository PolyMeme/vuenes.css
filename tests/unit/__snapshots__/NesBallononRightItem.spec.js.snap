// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesBalloonRightItem.vue should match snapshot 1`] = `
<div class="message -right">
  <div class="nes-balloon from-right">
    <p>This is a message from the right!</p>
  </div>
</div>
`;
